r/R/dplyr-functions.R

```diff
@@ -634,20 +634,45 @@ nse_funcs$wday <- function(x, label = FALSE, abbr = TRUE, week_start = getOption
 }
 
 nse_funcs$log <- function(x, base = exp(1)) {
-  
+
   if (base == exp(1)) {
     return(Expression$create("ln_checked", x))
   }
-  
+
   if (base == 2) {
     return(Expression$create("log2_checked", x))
   }
-  
+
   if (base == 10) {
     return(Expression$create("log10_checked", x))
-  } 
+  }
   # ARROW-13345
   stop("`base` values other than exp(1), 2 and 10 not supported in Arrow", call. = FALSE)
 }
 
 nse_funcs$logb <- nse_funcs$log
+
+nse_funcs$if_else <- function(condition, true, false, missing = NULL){
+  if (!is.null(missing)) {
+    arrow_not_supported("missing argument")
```

Comment:
What is this argument? Do we plan to support it in the future (i.e. is there a JIRA, or should there be)?

Reply:
It provides a value to be used if the `condition` is missing/NA:

```
> vec <- c(1, 2, NA, 3, 4)
> dplyr::if_else(vec < 4, "<4", ">=4", missing = "MISSING")
[1] "<4"      "<4"      "MISSING" "<4"      ">=4"    
```

Actually, we could support it now with a nested if_else... I'll work on that.